src/promptflow/promptflow/_sdk/_service/apis/flow.py

```diff
@@ -1,68 +1,170 @@
-    @api.response(code=200, description="Get flow snapshot", model=dict_field)
-    @api.doc(parser=flow_test_parser, description="Get flow snapshot")
+    @api.response(code=200, description="Return flow yaml as json", model=dict_field)
+    @api.doc(description="Return flow yaml as json")
     def get(self):
-
-        args = flow_test_parser.parse_args()
-        source_path = Path(args.flow)
-        flow_path = resolve_flow_path(source_path)
+        flow_path = api.payload["flow"]
+        flow_path = resolve_flow_path(Path(flow_path))
         flow_info = load_yaml(flow_path)
         return flow_info
+
+
+@api.route("/ux_inputs")
+class FlowUxInputs(Resource):
+    @api.response(code=200, description="Get the file content of file UX_INPUTS_JSON", model=dict_field)
+    @api.doc(description="Get the file content of file UX_INPUTS_JSON")
+    def get(self):
+        if not UX_INPUTS_JSON.exists():
+            UX_INPUTS_JSON.touch(mode=read_write_by_user(), exist_ok=True)
+        try:
+            ux_inputs = json_load(UX_INPUTS_JSON)
+        except json.decoder.JSONDecodeError:
+            ux_inputs = {}
+        return ux_inputs
+
+    @api.response(code=200, description="Set the file content of file UX_INPUTS_JSON", model=dict_field)
+    @api.doc(description="Set the file content of file UX_INPUTS_JSON")
+    def post(self):
+        content = request.get_json()
+        UX_INPUTS_JSON.touch(mode=read_write_by_user(), exist_ok=True)
+        json_dump(content, UX_INPUTS_JSON)
+        return make_response("UX_INPUTS_JSON content updated successfully", 200)
+
+
+def save_image(base64_data, extension, directory):
+    image_data = base64.b64decode(base64_data)
+    filename = str(uuid.uuid4())
+    file_path = os.path.join(directory, f"{filename}.{extension}")
+    with open(file_path, "wb") as f:
+        f.write(image_data)
+    return file_path
+
+
+@api.route('/image_save')
+class ImageSave(Resource):
+    @api.response(code=200, description="Save image", model=fields.String)
+    @api.doc(description="Save image")
+    @api.expect(image_save_model)
+    def post(self):
+        flow = api.payload["flow"]
+        base64_data = api.payload["base64_data"]
+        extension = api.payload["extension"]
+        file_path = save_image(base64_data, extension, flow)
+        return file_path
+
+
+@api.route('/image')
+class ImageUrl(Resource):
+    @api.response(code=200, description="Get image url", model=fields.String)
+    @api.doc(description="Get image url")
+    @api.expect(image_view_model)
+    def post(self):
+        image_path = api.payload["image_path"]
+        if not os.path.exists(image_path):
+            return make_response("The image doesn't exist", 404)
+        directory, filename = os.path.split(image_path)
+        directory = Path(directory).as_posix()
+        return f"{request.base_url}/{str(directory)}/{filename}"
+
+
+@api.route('/image/<path:directory>/<path:filename>')
+class ImageView(Resource):
```

Comment:
we can move these new apis for UI into a new route like 
/ui/image/...

Reply:
moved